dired.py

```diff
@@ -1364,6 +1365,44 @@ def vcs_check(self, path, git='git'):
             else:
                 self.vcs_state.update(git=False)
 
+    def call_hg(self, path):
+        hg = self.view.settings().get('hg_path', '')
+        if hg:  # empty string disable hg integration
+            self.hg_thread = threading.Thread(target=self.hg_check, args=(path, hg))
+            self.hg_thread.start()
+        else:
+            self.vcs_state.update(hg=False)
+
+    def hg_check(self, path, hg='hg'):
+        if any(c for c in '~*?[]' if c in hg):
+            match = glob.glob(os.path.expandvars(os.path.expanduser(hg)))
```

Comment:
It is last one, I promise :smile: 
Please, make change today, so I will focus on #91